models/classes/CompilationDataService.php

```diff
@@ -47,5 +48,51 @@ public function __construct($options = []) {
      * @return \qtism\data\QtiComponent
      * @throws \common_Exception In case of error.
      */
-    abstract public function readPhpCompilationData(\tao_models_classes_service_StorageDirectory $compilationDirectory, $path, $cacheInfo = '');
+    abstract public function readCompilationData(\tao_models_classes_service_StorageDirectory $compilationDirectory, $path, $cacheInfo = '');
+
+    /**
+     * Write Compilation Metadata
+     *
+     * @param \tao_models_classes_service_StorageDirectory $compilationDirectory
+     * @param AssessmentTest $test
+     * @throws \common_Exception
+     */
+    public function writeCompilationMetadata(\tao_models_classes_service_StorageDirectory $compilationDirectory, AssessmentTest $test)
+    {
+        try {
+            $filename = \taoQtiTest_models_classes_QtiTestService::TEST_COMPILED_META_FILENAME . '.json';
+            $meta = \taoQtiTest_helpers_TestCompilerUtils::testMeta($test);
+            $compilationDirectory->write($filename, json_encode($meta));
+        } catch (\Exception $e) {
+            throw new \common_Exception("Unable to write file '${filename}'.");
+        }
+
+    }
+
+    /**
+     * Read Compilation Metadata
+     *
+     * @param \tao_models_classes_service_StorageDirectory $compilationDirectory
+     * @return mixed
+     * @throws \common_Exception
+     */
+    public function readCompilationMetadata(\tao_models_classes_service_StorageDirectory $compilationDirectory)
+    {
+        try {
+            $data = $compilationDirectory->read(\taoQtiTest_models_classes_QtiTestService::TEST_COMPILED_META_FILENAME . '.json');
+            return json_decode($data, true);
+
+        } catch (\Exception $e) {
```

Comment:
Catching general `\Exception` seems to me not really reliable solution. Is that possible to get rid of all the legacy metadata php files by recompiling them into json during update?

Reply:
As we need a quick resolution, I don't think it is absolutely required at the present time. That can come in later on.